unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/Services/SceneControllerService/SceneControllerServiceImpl.cs

```diff
@@ -273,5 +277,184 @@ public async UniTask<CRDTSceneCurrentState> GetCurrentState(GetCurrentStateMessa
 
             return result;
         }
+
+        public async UniTask<SendBatchResponse> SendBatch(SendBatchRequest request, RPCContext context, CancellationToken ct)
+        {
+            await UniTask.SwitchToMainThread(ct);
+
+            try
+            {
+                foreach(var action in request.Actions)
+                    context.crdt.SceneController.EnqueueSceneMessage(new QueuedSceneMessage_Scene
+                    {
+                        type = QueuedSceneMessage.Type.SCENE_MESSAGE,
+                        method = MapMessagingMethodType(action),
+                        sceneNumber = sceneNumber,
+                        payload = ExtractPayload(from: action),
+                        tag = action.Tag,
+                    });
```

Comment:
@popuz you could use action directly, without converting it to "payload"

Reply:
for now we keep converting because not all messages are transferred to the new approach. We are still testing. But, yeah, I isolated this conversion part in recent PRs, so it is encapsulated in this ExtractPayload function and will be easier to remove